quickwit-actors/src/mailbox.rs

```diff
@@ -36,86 +39,102 @@ use crate::{Actor, AskError, Command, Handler, QueueCapacity, RecvError, SendErr
 ///
 /// The actor holds its `Inbox` counterpart.
 ///
-/// The mailbox can accept:
-/// - Regular messages wrapped in envelopes. Their type depend on the actor and is defined when
-/// implementing the actor trait. (See [`Envelope`])
-/// - Commands (See [`Command`]). Commands have a higher priority than messages:
-/// whenever a command is available, it is guaranteed to be processed
-/// as soon as possible regardless of the presence of pending regular messages.
+/// The mailbox can receive high priority and low priority messages.
+/// Commands are typically sent as high priority messages, whereas regular
+/// actor messages are sent to the low priority channel.
+///
+/// Whenever a high priority message is available, it is processed
+/// before low priority messages.
 ///
 /// If all mailboxes are dropped, the actor will process all of the pending messages
 /// and gracefully exit with [`crate::actor::ActorExitStatus::Success`].
 pub struct Mailbox<A: Actor> {
-    pub(crate) inner: Arc<Inner<A>>,
+    inner: Arc<Inner<A>>,
+    // We do not rely on the `Arc:strong_count` here to avoid an intricate
```

Comment:
Wow, I did not see this kind of issue could happen. Did you see or suspect weird behavior of Quickwit because of that?

Reply:
I spotted the bug with my little eyes.

I once notice strange behavior once but I don't know if it is related to this.